src/WebJobs.Script.WebHost/Swagger/SwaggerDocumentManager.cs

```diff
@@ -0,0 +1,168 @@
+                foreach (var httpOperation in httpOperations)
+                {
+                    // Trace is not recogized as an HttpMethod by swagger specification
+                    if (httpOperation.ToLower(CultureInfo.CurrentCulture) == HttpMethod.Trace.ToString().ToLowerInvariant())
+                    {
+                        continue;
+                    }
+
+                    HttpOperationInfo httpOperationInfo = new HttpOperationInfo()
+                    {
+                        InputParameters = inputParameters,
+                        OperationId = $"{endpoint}/{httpOperation}"
+                    };
+                    endpointsOperationData.Add(httpOperation, httpOperationInfo);
+                }
+            }
+            return apiEndpoints;
+        }
+
+        private static string[] GetHttpOperations(IHttpRoute httpRoute)
+        {
+            HttpMethodConstraint httpMethodConstraint = null;
+            httpRoute.Constraints.TryGetValue(ScriptConstants.HttpMethodConstraintName, out httpMethodConstraint);
+            if (httpMethodConstraint != null && httpMethodConstraint.AllowedMethods != null)
+            {
+                return httpMethodConstraint.AllowedMethods.Select(p => p.Method).ToArray<string>();
+            }
+            return ScriptConstants.HttpMethods.ToArray();
+        }
+
+        private static Collection<HttpOperationParameterInfo> GetInputParameters(IHttpRoute httpRoute)
+        {
+            var parameters = new Collection<HttpOperationParameterInfo>();
+            var pathParameters = GetAllPathParameters(httpRoute.RouteTemplate);
+            foreach (var constraint in httpRoute.Constraints)
+            {
+                var httpConstraint = constraint.Value as IHttpRouteConstraint;
+                SwaggerDataType? swaggerDataType = httpConstraint.ToSwaggerDataType();
+                if (swaggerDataType != null && httpConstraint.GetType() != typeof(HttpMethodConstraint))
+                {
+                    var parameter = new HttpOperationParameterInfo()
+                    {
+                        DataType = swaggerDataType.ToString().ToLowerInvariant(),
+                        Name = constraint.Key
+                    };
+                    parameters.Add(parameter);
+                    pathParameters.Remove(constraint.Key);
+                }
+            }
+
+            foreach (var pathParameter in pathParameters)
+            {
+                var parameter = new HttpOperationParameterInfo()
+                {
+                    DataType = SwaggerDataType.String.ToString().ToLowerInvariant(),
+                    Name = pathParameter
+                };
+                parameters.Add(parameter);
+            }
+            return parameters;
+        }
+
+        private static Collection<string> GetAllPathParameters(string routeTemplate)
+        {
+            var pathSegments = new Collection<string>();
```

Comment:
Can we make sure similar logic is not already exposed on the provided `IHttpRoute` or other routing types? We're already doing similar things to extract route data and ideally, we'd reuse that logic. We want to avoid this parsing logic if possible as there are always tricky cases and we want to ensure consistency.